sktime/distances/_twe.py

```diff
@@ -0,0 +1,315 @@
+            If the input time series do not have exactly 2 dimensions.
+            If the sakoe_chiba_window_radius is not an integer.
+            If the itakura_max_slope is not a float or int.
+        """
+        x = pad_ts(x)
+        y = pad_ts(y)
+        _bounding_matrix = resolve_bounding_matrix(
+            x, y, window, itakura_max_slope, bounding_matrix
+        )
+
+        @njit(cache=True)
+        def numba_twe_distance(
+            _x: np.ndarray,
+            _y: np.ndarray,
+        ) -> float:
+            cost_matrix = _twe_cost_matrix(_x, _y, _bounding_matrix, lmbda, nu, p)
+            return cost_matrix[-1, -1]
+
+        return numba_twe_distance
+
+
+@njit(cache=True)
+def pow_minkowski(x: np.ndarray, y: np.ndarray, p: int):
+    """Compute the powered Minkowski distance.
+
+    Parameters
+    ----------
+    x: np.ndarray (2d array of shape dxm1).
+        First time series.
+    y: np.ndarray (2d array of shape dxm1).
+        Second time series.
+    p: int
+        Order of the p-norm for local cost.
+
+    Returns
+    -------
+    float
+        Powered minkowski distance.
+    """
+    cost = np.sum(np.power(np.abs(x - y), p))
+    return np.power(cost, 1.0 / p)
+
+
+@njit(cache=True)
+def pad_ts(x: np.ndarray) -> np.ndarray:
+    """Pad the time with a 0.0 at the start.
+
+    Parameters
+    ----------
+    x: np.ndarray (of shape (d, m))
+        A time series.
+
+    Returns
+    -------
+    np.ndarray
+        A padded time series of shape (d, m + 1)
+    """
+    padded_x = np.zeros((x.shape[0], x.shape[1] + 1))
+    zero_arr = np.array([0.0])
+    for i in range(x.shape[0]):
+        padded_x[i, :] = np.concatenate((zero_arr, x[i, :]))
+    return padded_x
+
+
+@njit(cache=True)
+def _twe_cost_matrix(
+    x: np.ndarray,
+    y: np.ndarray,
+    bounding_matrix: np.ndarray,
+    lmbda: float,
+    nu: float,
+    p: int,
+) -> np.ndarray:
+    """Twe distance compiled to no_python.
+
+    Series should be shape (d, m), where d is the number of dimensions, m the series
+    length. Series can be different lengths.
+
+    Parameters
+    ----------
+    x: np.ndarray (2d array of shape dxm1).
+        First time series.
+    y: np.ndarray (2d array of shape dxm1).
+        Second time series.
+    bounding_matrix: np.ndarray (2d array of shape m1xm2)
+        Bounding matrix where the index in bound finite values (0.) and indexes
+        outside bound points are infinite values (non finite).
+    lmbda: float
+        A constant penalty that punishes the editing efforts. Must be >= 1.0.
+    nu: float
+        A non-negative constant which characterizes the stiffness of the elastic
+        TWE measure. Must be > 0.
+    p: int
+        Order of the p-norm for local cost.
+
+    Returns
+    -------
+    cost_matrix: np.ndarray (of shape (n, m) where n is the len(x) and m is len(y))
+        The dtw cost matrix.
+    """
+    x = pad_ts(x)
+    y = pad_ts(y)
```

Comment:
whats this for?

Reply:
In the pseudocode for the distance (given here: https://en.wikipedia.org/wiki/Time_Warp_Edit_Distance) and the implementations, I looked at (https://github.com/jzumer/pytwed) they all required you to pad the series so I did that as well.